packages/functional-tests/tests/subscription-tests/coupons-ui-tests/resubscription.spec.ts

```diff
@@ -0,0 +1,110 @@
+
+    const total = await subscribe.getTotalPrice();
+
+    //Subscribe successfully with Stripe
+    await subscribe.setFullName();
+    await subscribe.setCreditCardInfo();
+    await subscribe.clickPayNow();
+    await subscribe.submit();
+
+    //Login to FxA account
+    await login.goto();
+    await login.clickSignIn();
+    const subscriptionPage = await settings.clickPaidSubscriptions();
+    subscriptionManagement.page = subscriptionPage;
+
+    //Cancel subscription and then resubscribe
+    await subscriptionManagement.cancelSubscription();
+    await subscriptionManagement.resubscribe();
+
+    //Verify that the resubscription has the same coupon applied
+    expect(await subscriptionManagement.getResubscriptionPrice()).toEqual(
+      total
+    );
+  });
+
+  test('resubscribe successfully with the same coupon after canceling for paypal', async ({
+    page,
+    pages: { relier, subscribe, login, settings, subscriptionManagement },
+  }) => {
+    await relier.goto();
+    await relier.clickSubscribe6Month();
+
+    // 'auto10pforever' is a 10% forever discount coupon for a 6mo plan
+    await subscribe.addCouponCode('auto10pforever');
+
+    // Verify the coupon is applied successfully
+    expect(await subscribe.discountAppliedSuccess()).toBe(true);
+
+    const total = await subscribe.getTotalPrice();
+
+    //Subscribe successfully using Paypal
+    await subscribe.setPayPalInfo();
+    await subscribe.submit();
+
+    //Login to FxA account
+    await login.goto();
+    await login.clickSignIn();
+    const subscriptionPage = await settings.clickPaidSubscriptions();
+    subscriptionManagement.page = subscriptionPage;
+
+    //Cancel subscription and then resubscribe
+    await subscriptionManagement.cancelSubscription();
+    await subscriptionManagement.resubscribe();
+
+    //Verify that the resubscription has the same coupon applied
+    expect(await subscriptionManagement.getResubscriptionPrice()).toEqual(
+      total
+    );
+  });
+
+  test('update mode of payment for stripe', async ({
+    page,
+    pages: { relier, subscribe, login, settings, subscriptionManagement },
+  }) => {
+    await relier.goto();
+    await relier.clickSubscribe6Month();
+
+    // 'auto10pforever' is a 10% forever discount coupon for a 6mo plan
+    await subscribe.addCouponCode('auto10pforever');
+
+    //Subscribe successfully
+    await subscribe.setFullName();
+    await subscribe.setCreditCardInfo();
+    await subscribe.clickPayNow();
+    await subscribe.submit();
+
+    //Login to FxA account
+    await login.goto();
+    await login.clickSignIn();
+    const subscriptionPage = await settings.clickPaidSubscriptions();
+    subscriptionManagement.page = subscriptionPage;
+
+    //Change stripe card information
+    await subscriptionManagement.changeStripeCardDetails();
+
+    //Verify that the card info is updated
+    expect(await subscriptionManagement.getCardInfo()).toMatch(
+      'Card ending in ⁨4444⁩'
```

Comment:
"This line has hidden Unicode characters"
![Screenshot 2022-12-14 at 2 47 19 PM](https://user-images.githubusercontent.com/557895/207731711-e6d31174-ee0e-4022-95a7-b85f044b4eb2.png)

Reply:
removed it.